### PR TITLE
2594 feat: docs page

### PR DESCRIPTION
**Pull request by macintoshpie**
#### Any background context you want to provide?
SEED is preparing for a public release. Part of that effort is improving documentation.

#### What's this PR do?
Adds a public documentation page, which displays a list of frequently asked questions. Also, it updates the sidebar so that the hamburger menu is there instead of in the top navbar (see screenshots).

The current FAQ items were pulled directly from https://seed-platform.org/

#### How should this be manually tested?
Before testing, rerun pip install, this PR adds a python dependency.

Go to the documentation page (logged in and not) at `/documentation` or by clicking the "book" icon in the left sidebar. If not logged in, all of the navbar icons should be appropriately disabled.
You should see a list of faq categories, each expandable where you can then expand and view questions and their response by clicking on the question.
You should be able to use the search bar to filter by any word in the question/response, as well as by any hidden "tag" (tags are specified in the markdown frontmatter, in the docs/faq/<category>/<faq item>.md file.

You can also try creating new markdown files and see how they're added to the list of FAQ items.

#### What are the relevant tickets?
#2594 

#### Screenshots (if appropriate)
Screen Shot 2021-02-26 at 1 21 16 PM
Screen Shot 2021-02-26 at 1 21 36 PM
Screen Shot 2021-02-26 at 1 21 54 PM

### CODE REVIEW

File: seed/docs/templates/docs/faq.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{% extends "seed/base.html" %}{% load compress %}
```
Comment (macintoshpie) [inline seed/docs/templates/docs/faq.html:1]: note that we're using the same angular SEED app as registered at `/app/`. Kind of strange, but our only option for now due to current authentication requirements for other django views

File: seed/docs/templates/docs/faq.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    
+    {% include "seed/_sidebar.html" %}
+    
+    <div class="display">
+        <div ng-class="{narrow: narrow_controller, wide: wide_controller, content: true, hide_transition:is_initial_state}">
+            
+            {% include "seed/_header.html" %}
+            
+            <div class="page">
+                {{ faq_data|json_script:"faq-data" }}
```
Comment (macintoshpie) [inline seed/docs/templates/docs/faq.html:17]: safe way to inject JSON into template. https://docs.djangoproject.com/en/3.1/ref/templates/builtins/#json-script

File: seed/static/seed/js/controllers/menu_controller.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ angular.module('BE.seed.controller.menu', [])
       $scope.narrow_controller = false;
       $scope.wide_controller = false;
       $scope.username = window.BE.username;
+      $scope.logged_in = $scope.username.length > 0;
```
Comment (macintoshpie) [inline seed/static/seed/js/controllers/menu_controller.js:43]: By adding this we can render the sidebar and topbar without the user being logged in

File: seed/templates/seed/_sidebar.html [hunk 4]
```diff
@@ -1,41 +1,67 @@
...
+        <div class="divider"></div>
+        <a id="sidebar-profile" class="item" ng-class="{'disabled-item': !logged_in}" ui-sref="profile" ui-sref-active="active">
             <div class="icon"><i class="fa fa-cog"></i></div>
             <div ng-show="!is_initial_state" class="item_name">{$ username $}</div>
         </a>
         <div class="divider"></div>
-        <a id="sidebar-inventory" ng-class="{item: true, active: is_active('/properties') || is_active('/taxlots')}" href="{% url "seed:home" %}#/properties">
-            <div class="icon"><img src="{{ STATIC_URL }}seed/images/icon-city.svg" title="{$:: 'Inventory' | translate $}"></div>
+        <a id="sidebar-inventory" ng-class="{item: true, active: is_active('/properties') || is_active('/taxlots'), 'disabled-item': !logged_in}" href="{% url "seed:home" %}#/properties">
+            <div class="icon">
```
Comment (macintoshpie) [inline seed/templates/seed/_sidebar.html:17]: had to directly insert svg path here b/c we need to change the fill color using css

**Comment by RDmitchell** [comment]
I would be interested in reviewing the content before this is released. 

In particular, I would like to review the content of the linking, merging, mapping, etc

**Comment by macintoshpie** [comment]
@RDmitchell The questions/answers are written in markdown, are you comfortable reading it like this and providing feedback (you can view/comment on the files [here](https://github.com/SEED-platform/seed/pull/2604/commits/cdfe55b3b2d061c119611375f4f8b4f34663dac8))? Alternatively we can merge/deploy this to dev1, have you review there, then we can update as needed.

File: seed/static/seed/js/seed.js [hunk 5]
```diff
@@ -99,7 +99,8 @@ angular.module('BE.seed.controllers', [
   'BE.seed.controller.show_populated_columns_modal',
   'BE.seed.controller.ubid_modal',
   'BE.seed.controller.unmerge_modal',
-  'BE.seed.controller.update_item_labels_modal'
+  'BE.seed.controller.update_item_labels_modal',
+  'BE.docs.controller.faq'
```
Comment (adrian-lara) [inline seed/static/seed/js/seed.js:103]: Sorry - nitpicking here 😬  Think we could stick to keeping this alphabetized?

File: seed/templates/seed/_scripts.html [hunk 6]
```diff
@@ -124,6 +124,7 @@
         <script src="{{STATIC_URL}}seed/js/utilities/spinner.js"></script>
 
         {% endcompress %}
+        <script src="{{STATIC_URL}}docs/js/faq.js"></script>
```
Comment (adrian-lara) [inline seed/templates/seed/_scripts.html:127]: Can/should this exist within the compress block?

File: seed/docs/views.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+from collections import namedtuple
+import os
+import re
+
+import markdown
+import yaml
+
+
+YAML_DOC_BOUNDARY = re.compile(r"^-{3,}\s*$", re.MULTILINE)
```
Comment (adrian-lara) [inline seed/docs/views.py:17]: Would there be any issue with parsing multiple `-`'s used for title headers in the markdown section?

**Review by adrian-lara — COMMENTED** [review]
This works as expected. A few code questions but this largely looks good! 

It's worth getting another set of eyes on the more notable templating code changes - primarily `seed/docs/templates/docs/faq.html` and `seed/templates/seed/_sidebar.html` and the comments within each. @axelstudios

File: seed/docs/views.py (hunk 7 above)
Comment (macintoshpie) [reply seed/docs/views.py:17]: Shouldn't be a problem, the way we use `split()` just splits on the first two occurrences:
https://github.com/SEED-platform/seed/blob/109d509c5131eac823f8f64921ce7e6bcdc3bb70/seed/docs/views.py#L40

ref: https://docs.python.org/3/library/re.html#re.split

**Comment by axelstudios** [comment]
Just fixed several small issues, looks good to me:

## Fixed issues
- The documentation sidebar icon doesn't show as active
- If you're logged in and click the "gear" profile button it will fail
- If you're logged in there will be a failed 400 request for `http://seed.lan:8000/api/v3/datasets/count/?organization_id=`
- If you're logged in the organization dropdown appears but is empty
- If you're logged in and click the "+" in the top right corner to create a dataset it fails
- If you're logged in the datasets in the sidebar show 0
- If you're not logged in both orgs and datasets in the sidebar show 0

## Improvements
- Added autofocus to the input
- Added one-time binding to the html
- Fixed some of the jumping while expanding by using $timeout to run after evaluating the ng-show first
- I put the sidebar svg back as an img and used opacity rather than color/fill

**Review by axelstudios — APPROVED** [review]

File: seed/static/seed/js/controllers/menu_controller.js [hunk 8]
```diff
@@ -66,21 +66,16 @@ angular.module('BE.seed.controller.menu', [])
...
+        if (menu_item === current_path) {
           return true;
-        } else if (menu_item === '/seed/data' && !_.includes($location.absUrl(), '#')) {
-          if (_.includes($location.absUrl(), menu_item)) return true;
-          if (_.includes($location.absUrl(), 'worksheet')) return true;
-          if (_.includes($location.absUrl(), 'mapping')) return true;
-          if (_.includes($location.absUrl(), 'cleaning')) return true;
-          if (_.includes($location.absUrl(), 'merge')) return true;
-          if (_.includes($location.absUrl(), 'import')) return true;
-          return false;
```
Comment (macintoshpie) [inline seed/static/seed/js/controllers/menu_controller.js:80]: 💣💥

File: seed/static/seed/scss/style.scss [hunk 9]
```diff
@@ -271,11 +271,9 @@ a:not([href]) {
         display: inline;
       }
 
-      &.disabled-item,
-      &.disabled-item path {
+      &.disabled-item {
+        opacity: 0.07;
```
Comment (macintoshpie) [inline seed/static/seed/scss/style.scss:275]: smart!

**Comment by macintoshpie** [comment]
Reviewed Alex's changes and tested locally, things work as expected. Thanks for the help @axelstudios !

File: seed/static/seed/js/controllers/menu_controller.js (hunk 8 above)
Comment (axelstudios) [reply seed/static/seed/js/controllers/menu_controller.js:80]: Yep, that stuff was vestigial

**Pull request merged**